pkgs/development/libraries/frei0r/default.nix

```diff
@@ -1,26 +1,16 @@
-{ lib, stdenv, fetchurl, fetchpatch, cairo, cmake, opencv, pcre, pkg-config }:
+{ lib, config, stdenv, fetchurl, cairo, cmake, opencv, pcre, pkg-config, cudaSupport ? (config.cudaSupport or false) && stdenv.hostPlatform.isx86_64, cudaPackages ? { } }:
 
 stdenv.mkDerivation rec {
   pname = "frei0r-plugins";
-  version = "1.7.0";
+  version = "1.8.0";
 
   src = fetchurl {
     url = "https://files.dyne.org/frei0r/releases/${pname}-${version}.tar.gz";
-    hash = "sha256-Gx/48Pm8I+7XJOlOmnwdjwJEv+M0JLtP5o5kYMCIUjo=";
+    hash = "sha256-RaKGVcrwVyJ7RCuADKOJnpNJBRXIHiEtIZ/fSnYT9cQ=";
   };
 
-  # A PR to add support for OpenCV 4 was merged in May 2020. This
-  # patch can be removed when a release beyond 1.7.0 is issued.
-  patches = [
-    (fetchpatch {
-      name = "opencv4-support.patch";
-      url = "https://github.com/dyne/frei0r/commit/c0c8eed79fc8abe6c9881a53d7391efb526a3064.patch";
-      sha256 = "sha256-qxUAui4EEBEj8M/SoyMUkj//KegMTTT6FTBDC/Chxz4=";
-    })
-  ];
-
   nativeBuildInputs = [ cmake pkg-config ];
-  buildInputs = [ cairo opencv pcre ];
+  buildInputs = [ cairo opencv pcre (lib.optionals cudaSupport [ cudaPackages.cudatoolkit ]) ];
```

Comment:
I'd ask that we do this using the redist cuda packages (`cuda_cudart`, ...) but `opencv` currently uses `cudatoolkit` anyway, and maybe we should prepare the setup hooks for `FindCUDAToolkit.cmake` first

Reply:
What do you mean exactly with preparing setup hooks for `FindCUDAToolkit.cmake`?
I use `cudaPackages.cuda_cudart` and `cudaPackages.cuda_nvcc` now instead of `cudaPackages.cudatoolkit`.

Reply:
Oh, no matter, I just wasn't sure if it's going to work out of the box. It's great to see that it has!